pkg/features/kube_features.go

```diff
@@ -793,7 +793,7 @@ const (
 	// alpha: v1.25
 	//
 	// Enables user namespace support for stateless pods.
-	UserNamespacesStatelessPodsSupport featuregate.Feature = "UserNamespacesStatelessPodsSupport"
```

Comment:
This is effectively a breaking change.  Can we make this add a new gate and then support both as equivalents?  Or only support stateless if stateless gate is set and stateless + stateful in new gate?

Reply:
I'm not sure that is a good idea, the user can have some surprises if we do that. Allow me to elaborate.

If the `UserNamespacesStatelessPodsSupport` feature gate also enables stateful pods, then the user can _very_ easily be surprised about that and is not something intuitive. Furthermore, as the requirements for stateless pods also changed (kernel support, container runtime support also needs changes, etc.) it might even be the case that with the stack they are using userns is not supported for stateless pods either. So it is not, if the feature gate does that, then things will "just work" for users of stateless pods.

Doing only stateless if only the stateless feature gate is active is tricky and not sure the churn is worth it. Because of what I already mentioned (things will "just work" for users of stateless pods) and because we will need to make the validation conditional on a feature gate. That is tricky, as we have the same pod.spec field name for stateless and stateful (pod.spec.hostUsers), and we can't check for the feature gate in that part of the code: https://github.com/kubernetes/kubernetes/pull/111090#discussion_r933556896, we might need to modify the ForUpdate/similar functions, if that is even possible to do.

If there were other reasons, it can be worth exploring that path. But IMHO, the pain is not worth the effort. This is an alpha feature, that was reworked and has new requirements from the container runtime and the kernel now, let's take this chance to change the feature gate and make sure users that opt-in are aware of the new requirements introduced.

@thockin If you feel strongly about doing any of those things, we can of course do it. In the end we want this to progress, so if you feel strongly about it, we will of course do it. But wanted to highlight those things in case you weren't aware

Reply:
OK.  If we release-note it as "alpha feature gate UserNamespacesStatelessPodsSupport has been abandoned" and "alpha feature gate UserNamespaces is new", I guess I can live with that.